src/device.ts

```diff
@@ -312,15 +312,24 @@ export class Device extends vscode.Disposable {
      */
     public async mkdir_p(path: string): Promise<void> {
         const names = path.split('/');
+
+        // If path had a preceding slash, remove the empty path segment
+        if(!names[0]) {
+            names.shift();
+        }
+
         let part = '';
         while (names.length) {
-            part += names.shift() + '/';
-            // have to make sure the directory exists on the remote device first
+            part += '/' + names.shift();
```

Comment:
I just realized a day or two ago that there is a `path.posix` namespace and get unix separators even on Windows. So, in places where the code is doing `+ '/'` type things, I would like to use `path.posix.join()` instead (if we happen to be touching that code anyway).

Reply:
You mean, `path.posix.join('', names.shift())`? Or instead use `path.posix.separator`? Also, it'd be helpful to avoid using module names as variable names in the future; there are a bunch of places where we have variables called `path` which makes this a slightly larger change.

Reply:
I was thinking

   part = path.posix.joint(part, names.shift());

We can save cleaning up variables named `path` for another PR.

Reply:
Oh, got it.

> We can save cleaning up variables named path for another PR.

Agreed, I'll need to change this instance though.